src/lein_monolith/plugin.clj

```diff
@@ -152,23 +175,33 @@
 
 ;; ## Plugin Middleware
 
+(defn- add-profiles
+  "Adds profiles to the project. Profiles should be passed in as a vector to ensure profiles are
+   added in the right order."
+  [project profiles]
+  (reduce (fn [project [profile-key profile]]
```

Comment:
Building on the "merge-profiles is expensive" note below, it'd be good to check if `profiles` is non-empty here before we do this, and return the `project` unchanged if not.